### ISSUE TITLE
lfda not deterministic

### ISSUE THREAD

**Issue by wdevazelhes** [scikit-learn-contrib/metric-learn#211]
See comment https://github.com/metric-learn/metric-learn/pull/193#pullrequestreview-245994079
It seems that `LFDA` is not deterministic because of an eigenvalue decomposition that can return one sign or another as output
More details to come

**Comment by bellet** [comment]
I think it is fine, this happens also in sklearn, see for instance the "Notes" section:
https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.TruncatedSVD.html

What we could do however is to have a random_state in init (related PR: #171) so that we two instances of LFDA with the same random state and fit once on the same data will give the same result
